### PR TITLE
[Debt] Migrate IAP application test to playwright

### LINKED ISSUE #10495: ♻️ Move e2e to Playwright - submit-application-iap.cy.ts
## ♻️ Debt/Refactor

Time to move our Cypress tests to Playwright
apps/e2e/cypress/e2e/talentsearch/submit-application-iap.cy.ts

## ✅ Acceptance Criteria

- [ ] Test moved to Playwright : submit-application-iap.cy.ts

## 🛑 Blockers

```[tasklist]
### Blocked By
```

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #10495

## 👋 Introduction

Migrates the IAP application test to playwright.

## 🧪 Testing

1. Confirm same coverage
2. Confirm test passes

### CODE REVIEW

File: apps/playwright/tests/applicant/application-iap.spec.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    await application.page
+      .getByRole("link", { name: /apply now/i })
+      .first()
+      .click();
+    await application.waitForGraphqlResponse("Application");
+
+    // Welcome page - step one
+    await expectOnStep(application.page, 1);
+    await application.page.getByRole("button", { name: /let's go/i }).click();
+    // await application.waitForGraphqlResponse("Application");
```
Comment (vd1992) [inline apps/playwright/tests/applicant/application-iap.spec.ts:134]: To delete or uncomment?

**Review by vd1992 — COMMENTED** [review]
Works for me part of the time, the tests are a bit flaky with timeouts across the three browser. Let us have someone else run it too

**Comment by vd1992** [comment]
For reference, did a bunch more runs. Clear hierarchy in flakiness 😄 

Where Firefox runs fail, one fail in four runs

image

Where Webkit runs fail, three fail in four runs

image

image

Where Chromium runs fail, zero fail in four runs

N/A

**Comment by esizer** [comment]
> For reference, did a bunch more runs. Clear hierarchy in flakiness 😄

Are you reseeding your data in between tests? My guess is you have a user create an experience but then that experience ecists so it fails on the second run

**Comment by vd1992** [comment]
> > For reference, did a bunch more runs. Clear hierarchy in flakiness 😄
> 
> Are you reseeding your data in between tests? My guess is you have a user create an experience but then that experience ecists so it fails on the second run

No I am not.
I don't think that makes a difference, I ran the Chromium runs consecutively after the other browser workflows but they all passed

**Comment by esizer** [comment]
> No I am not.
> I don't think that makes a difference, I ran the Chromium runs consecutively after the other browser workflows but they all passed

So you are doing:

1. `make seed-fresh`
2. `pnpm run e2e:playwright:chromium` - Passes
3. `pnpm run e2e:playwright:chromium` - Passes
4. `make seed-fresh`
5. `pnpm run e2e:playwright:firefox` - Fails

**Comment by vd1992** [comment]
I did 

`pnpm run e2e:playwright:firefox` x4
then
`pnpm run e2e:playwright:webkit` x4
then
`pnpm run e2e:playwright:chromium` x4

To do the above, no re-seeding anywhere

**Comment by vd1992** [comment]
Trying Webkit with re-seed

1. `make seed-fresh`
2. `pnpm run e2e:playwright:webkit` passes
3. `make seed-fresh`
4. `pnpm run e2e:playwright:webkit` failed

image

**Comment by esizer** [comment]
> Trying Webkit with re-seed
> 
> 1. `make seed-fresh`
> 2. `pnpm run e2e:playwright:webkit` passes
> 3. `make seed-fresh`
> 4. `pnpm run e2e:playwright:webkit` failed
> 
> image

Have you tried increasing your timeouts or running with a single worker? I cannot reproduce this which makes me think the 10 second timeout is just too low. Also, it is hard to troubleshoot without traces. I assume when you check the trace the page just hasn't loaded yet after the 10 seconds so it just gives up (which is quite common running locally).

If this is the case, you can find a decent time out and just set it in the [env file](https://github.com/GCTC-NTGC/gc-digital-talent/blob/ea54c9662c974183d60e53931bd38cbd99c811d1/apps/playwright/.env.example#L4) and then uncomment the loading of it when you run tests locally.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/ea54c9662c974183d60e53931bd38cbd99c811d1/apps/playwright/playwright.config.ts#L3-L7

**Comment by mnigh** [comment]
> > Trying Webkit with re-seed
> > 
> > 1. `make seed-fresh`
> > 2. `pnpm run e2e:playwright:webkit` passes
> > 3. `make seed-fresh`
> > 4. `pnpm run e2e:playwright:webkit` failed
> > 
> 
> Have you tried increasing your timeouts or running with a single worker? I cannot reproduce this which makes me think the 10 second timeout is just too low. Also, it is hard to troubleshoot without traces. I assume when you check the trace the page just hasn't loaded yet after the 10 seconds so it just gives up (which is quite common running locally).

@vd1992 this has been done in the playwright config in #10576 (and `main` has been merged into this PR's branch) so there shouldn't be a need to manually set this.

**Review by yonikid15 — COMMENTED** [review]
Mines passing for Chrome and Firefox, but failing on webkit. However, webkit doesn't seem to work on the majority of tests. I increased the timeouts as well still no luck with webkit

image

**Comment by esizer** [comment]
> Mines passing for Chrome and Firefox, but failing on webkit. However, webkit doesn't seem to work on the majority of tests. I increased the timeouts as well still no luck with webkit

Webkit fails 100% of the time for me as well as a false positive. If you look at the trace the error it throws is not actually an error :confounded:

**Comment by vd1992** [comment]
Here is the trace for the Firefox flakiness I get sometimes, it appears it asserts text appearance while still on the wrong page

image

**Review by vd1992 — APPROVED** [review]
Well that was a doozy

**Pull request merged**